FontConverter restore order of font properties

RussKie opened the pull request:
Fixes #45631

## Proposed changes:

.NET Framework implementation of `FontConverter` provided a different order of font properties, that can be observed by inspecting an instance of a font in a `PropertyGrid`.

![image](https://user-images.githubusercontent.com/4403806/59329728-f207f600-8d32-11e9-81f3-80cd6c2ae893.png)


Restore the original sort order.

The .NET Framework implementation has another sort argument "Weight", which however doesn't appear to be a property of `Font` type. It is possible this property has existed at some point, or the converter may have been expected to work for instances of `IFontDisp` object. Either way presence or absence of "Weight" doesn't appear to make any
difference, hence it has not been ported across.



## Testing strategy

- Added unit test
- Manual testing (by referencing .\artifacts\packages\Debug\Shipping\System.Drawing.Common.6.0.0-dev.nupkg)

## Screenshots 

* .NET Framework
    - en-AU culture
![image](https://user-images.githubusercontent.com/4403806/101580092-6d349000-3a2e-11eb-9b2e-3d7509de0955.png)
    - ru-RU culture
![image](https://user-images.githubusercontent.com/4403806/101579660-4d04d100-3a2e-11eb-90b7-2e4ff63a4a3a.png)

* .NET 5.0
    - en-AU culture
![image](https://user-images.githubusercontent.com/4403806/101581894-ed5af580-3a2e-11eb-8f96-9c59642904ae.png)


* With the fix
    - en-AU culture
![image](https://user-images.githubusercontent.com/4403806/101572862-44ab9680-3a2c-11eb-9c47-57d9412d95ca.png)
    - ru-RU culture
![image](https://user-images.githubusercontent.com/4403806/101573828-a409a680-3a2c-11eb-87ea-3cdd6d195de9.png)

RussKie committed:
FontConverter restore order of font properties

.NET Framework implementation of `FontConverter` provided a different
order of font properties, that can be observed by inspecting an instance
of a font in a `PropertyGrid`.

Restore the original sort order.

The .NET Framework implementation has another sort argument "Weight",
which however doesn't appear to be a property of `Font` type.
It is possible this property has existed at some point, or the converter
may have been expected to work for instances of `IFontDisp` object.
Either way presence or absence of "Weight" doesn't appear to make any
difference, hence it has not been ported across.

Fixes #45631

tannergooding commented:
Is the .NET framework order "common" (Name/Size/Unit) first and then alphabetical for everything else?

tannergooding commented:
Is this impacted at all by localization? Are there any UX or accessibility concerns/breaks with the change?
Are there any other properties that don't follow the alphabetical ordering that looks to be the current default?

RussKie commented:
> Is the .NET framework order "common" (Name/Size/Unit) first and then alphabetical for everything else?

Yes. Added screenshots from .NET Framework 4.7.2





> Is this impacted at all by localization? Are there any UX or accessibility concerns/breaks with the change?

I don't believe so, as the runtime isn't being localised. 
I've tested in en-AU and ru-RU locales to confirm, added screenshots from .NET Framework and .NET 6.0. 

> Are there any other properties that don't follow the alphabetical ordering that looks to be the current default?

At this stage I'm not aware of any, we haven't seen any other reports.

safern commented:
Merging and if @tannergooding has more follow up questions/concerns, we can always put up a follow up PR.

RussKie commented:
Thank you
